pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -279,9 +279,9 @@ func (r *Reconciler) gcResourceTrackers(logCtx monitorContext.Context, handler *
 	}
 	logCtx.Info("GarbageCollected resourcetrackers")
 	if phase == common.ApplicationRendering {
-		return ctrl.Result{}, r.updateStatus(logCtx, handler.app, common.ApplicationRunningWorkflow)
+		return ctrl.Result{RequeueAfter: common2.ApplicationReSyncPeriod}, r.updateStatus(logCtx, handler.app, common.ApplicationRunningWorkflow)
 	}
-	return ctrl.Result{}, r.patchStatus(logCtx, handler.app, phase)
```

Comment:
can we check whether we have changed the status? if changed, we shouldn't add a requeue.

Reply:
Yes. There should be some better check here.